axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/StockMoveInvoiceServiceImpl.java

```diff
@@ -204,25 +204,25 @@ public Invoice createInvoiceFromSaleOrder(
           invoice.setNote(invoice.getCompanyBankDetails().getSpecificNoteOnInvoice());
         }
       }
-      
+
       if (ObjectUtils.isEmpty(invoice.getProformaComments())) {
-          if (!Strings.isNullOrEmpty(saleOrder.getProformaComments())
-              && invoice.getCompanyBankDetails() != null
-              && !Strings.isNullOrEmpty(invoice.getCompanyBankDetails().getSpecificNoteOnInvoice())) {
-            invoice.setProformaComments(
-                saleOrder.getProformaComments()
-                    + "\n"
-                    + invoice.getCompanyBankDetails().getSpecificNoteOnInvoice());
-          } else if (Strings.isNullOrEmpty(saleOrder.getProformaComments())
-              && invoice.getCompanyBankDetails() != null
-              && !Strings.isNullOrEmpty(invoice.getCompanyBankDetails().getSpecificNoteOnInvoice())) {
-            invoice.setProformaComments(saleOrder.getProformaComments());
-          } else if (!Strings.isNullOrEmpty(saleOrder.getProformaComments())
-              && invoice.getCompanyBankDetails() != null
-              && Strings.isNullOrEmpty(invoice.getCompanyBankDetails().getSpecificNoteOnInvoice())) {
-            invoice.setProformaComments(invoice.getCompanyBankDetails().getSpecificNoteOnInvoice());
-          }
+        if (!Strings.isNullOrEmpty(saleOrder.getProformaComments())
+            && invoice.getCompanyBankDetails() != null
+            && !Strings.isNullOrEmpty(invoice.getCompanyBankDetails().getSpecificNoteOnInvoice())) {
+          invoice.setProformaComments(
+              saleOrder.getProformaComments()
+                  + "\n"
+                  + invoice.getCompanyBankDetails().getSpecificNoteOnInvoice());
+        } else if (Strings.isNullOrEmpty(saleOrder.getProformaComments())
+            && invoice.getCompanyBankDetails() != null
+            && !Strings.isNullOrEmpty(invoice.getCompanyBankDetails().getSpecificNoteOnInvoice())) {
+          invoice.setProformaComments(saleOrder.getProformaComments());
+        } else if (!Strings.isNullOrEmpty(saleOrder.getProformaComments())
+            && invoice.getCompanyBankDetails() != null
+            && Strings.isNullOrEmpty(invoice.getCompanyBankDetails().getSpecificNoteOnInvoice())) {
+          invoice.setProformaComments(invoice.getCompanyBankDetails().getSpecificNoteOnInvoice());
         }
+      }
 
```

Comment:
Please put this duplicated block in a new method and call it from both services.